dev/tasks/r/github.packages.yml

```diff
@@ -255,12 +253,7 @@ jobs:
           - {image: "rocker/r-ver:4.0.0"} # ubuntu-20.04
           - {image: "rocker/r-ver:3.6.3", libarrow_binary: "TRUE"} # debian:buster (10)
           - {image: "rstudio/r-base:4.1-bionic"} # ubuntu-18.04
-          # TODO(ARROW-17594): re-enable this with the devtoolset for C++17
-          #- {image: "rstudio/r-base:4.2-centos7"}
-          # Even with devtoolset-8, binaries fail because glibc is older than
-          # what we build with on ubuntu-18.04 (uncomment to see the error)
-          # TODO(ARROW-16976): build binaries with older glibc
-          # - {image: "rstudio/r-base:4.2-centos7", devtoolset: "8"}
+          - {image: "rstudio/r-base:4.2-centos7", devtoolset: "8"}

```

Comment:
Remove devtoolset here so we get the test you did manually of using with gcc 4.8? (How does that work at all given that we require C++17 now?)

Reply:
IIUC devtoolset statically links in the parts of the libc that are missing on the older versions. 

Good idea, I will add this as another step because the package also needs to be built with dts as we use some c++17 features (e.g. std::optional).

Reply:
Oh right, got it. This is the one that does `R CMD INSTALL --build`, so it is building the package but not libarrow, so it still needs devtoolset for C++17. But if you take the binary artifact from that, you can install it without devtoolset?

Would be good to document this and add messaging in `configure` or `build_arrow_static.sh` so that if someone is trying to install from source with gcc 4.8, we tell them what they need to do.

Reply:
https://issues.apache.org/jira/browse/ARROW-17849 :smile: 

> add messaging in configure or build_arrow_static.sh so that if someone is trying to install from source with gcc 4.8, we tell them what they need to do.

Oh good point! I'll add that to the jira

Reply:
Works! https://github.com/ursacomputing/crossbow/actions/runs/3158781948/jobs/5141426068